src/main/webapp/profile.js

```diff
@@ -13,30 +13,52 @@
 // limitations under the License.
 
 function onProfileLoad() {
+  supplyLogoutLink();  
+  autofillForm(getUserIfRegistered());  
```

Comment:
I don't think this works the way you think it does.

getUserIfRegistered returns nothing here, but even if you were returning the promise, it would not be accessible as if it was a person. You need to chain a `.then()` off of it.